clean_project/index.py

```diff
@@ -31,17 +29,48 @@ def index(config_path, index_path):
                         index_words[word.lower()][source_file.name].append(i)
     else:
         raise ValueError('Empty configuration file')
-    with open(index_path, 'w') as index_file:
-        json.dump(index_words, index_file, sort_keys=True, indent=4)
     return index_words
 
 
+def build_index(index_words):
+    """
+    Insert the index in a specific database for store all data
+    :param index_words: Object which contains the built index
+    """
+    connection = None
+    try:
+        connection = psycopg2.connect(database='index_db', user='gianluca', password='password')
+        cursor = connection.cursor()
+        query_update = "UPDATE index SET occurrences=%s WHERE word=%s AND file=%s;"
+
+        query_insert = ("INSERT INTO index (word, file, occurrences)"
+                        "SELECT %s, %s, %s"
+                        "WHERE NOT EXISTS (SELECT 1 FROM index WHERE word=%s AND file=%s);")
+        for word in index_words:
+            for file_path in index_words[word]:
+                data_update = (index_words[word].get(file_path), word, file_path)
```

Comment:
As described in official documentation, fortunately we have no need to do nothing about transaction: 
"In Psycopg transactions are handled by the connection class. By default, the first time a command is sent to the database (using one of the cursors created by the connection), a new transaction is created. The following database commands will be executed in the context of the same transaction – not only the commands issued by the first cursor, but the ones issued by all the cursors created by the same connection. Should any command fail, the transaction will be aborted and no further command will be executed until a call to the rollback() method."